Bug: Aplicação fica na fila de processamento para erros 301

@tiagoben, bom dia!

Tudo bem?

Com base no último [merge efetuado](https://github.com/tst-labs/esocial/pull/433) no branch principal, descobri um possível bug (não sei se é um bug ou se é intencional), que impede de a fila, que está em PROCESSAMENTO, continuar o processamento.

Debugando esse método [preencherProcessamentoEnviosEvento](https://github.com/tst-labs/esocial/blob/master/src/esocial-jt-service/src/main/java/br/jus/tst/esocialjt/negocio/AtualizacaoProcessamentoServico.java#L103), percebi que um dos eventos de envio de um determinado lote está com esse código **301**, conforme a imagem abaixo:

![Image](https://github.com/user-attachments/assets/25347a16-1056-4e49-9a4b-ce81784f5465)

Pegando a mensagem do erro, no campo **descricaoRespostaProcessamento**, está assim: 

> Código` do erro: 301.3. Hash: <HASH_XXXXXXXX>. Identificador: <IDENTIFICADOR_XXXXXXX>. A solicitação não pode ser atendida devido a uma falha não identificada que pode ou não ser temporária. Favor tentar novamente mais tarde. Caso o erro permaneça por um período superior a 6 (seis) horas, favor acessar o Portal do eSocial através do endereço https://www.gov.br/esocial/pt-br/canais_atendimento. Na opção CONTATO, na seção EMPRESAS, selecione PRODUÇÃO EMPRESAS. Preencha o formulário sobre erro 301 informando o código HASH do retorno e o identificador do evento para rastreamento. Obrigado.

Pelo que percebi, temos que reportar ao portal eSocial, conforme o que está [no site](https://www.gov.br/esocial/pt-br/noticias/erro-301-o-que-fazer)

Por causa desse erro, o **nrRecibo** está vindo null, conforme o log abaixo, ao tentar salvar o [Totalizador](https://github.com/tst-labs/esocial/blob/master/src/esocial-jt-service/src/main/java/br/jus/tst/esocialjt/negocio/AtualizacaoProcessamentoServico.java#L86) na base de dados.

```bash
Caused by: org.postgresql.util.PSQLException: ERROR: insert or update on table "est_evento_totalizador" violates foreign key constraint "est_evento_totalizador_fk01"
  Detail: Key (txt_nr_recibo_arq_base)=(XXXXXXXXXXXXXXXX) is not present in table "est_evento".
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2674) ~[postgresql-42.3.1.jar!/:42.3.1]
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2364) ~[postgresql-42.3.1.jar!/:42.3.1]
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:354) ~[postgresql-42.3.1.jar!/:42.3.1]
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:484) ~[postgresql-42.3.1.jar!/:42.3.1]
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:404) ~[postgresql-42.3.1.jar!/:42.3.1]
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:162) ~[postgresql-42.3.1.jar!/:42.3.1]
	at org.postgresql.jdbc.PgPreparedStatement.executeUpdate(PgPreparedStatement.java:130) ~[postgresql-42.3.1.jar!/:42.3.1]
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61) ~[HikariCP-4.0.3.jar!/:na]
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java) ~[HikariCP-4.0.3.jar!/:na]
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:197) ~[hibernate-core-5.6.1.Final.jar!/:5.6.1.Final]
	... 74 common frames omitted
```

**A perguta é**: É intencional deixar esses eventos em fila ou realmente deveria ser marcado como um erro visível para o usuário, já que só descobri debugando o código? 

Existe alguma solução de contorno?

Não seria melhor "cuspir" esse erro no painel para o usuário saber o que fazer ?

Obrigado!

--- a/src/esocial-jt-service/src/main/java/br/jus/tst/esocialjt/dominio/EventoTotalizador.java
+++ b/src/esocial-jt-service/src/main/java/br/jus/tst/esocialjt/dominio/EventoTotalizador.java
@@ -141,4 +141,18 @@ public class EventoTotalizador implements Serializable {
 		this.evento = evento;
 	}
 
+	@Override
+	public String toString() {
+		final StringBuffer sb = new StringBuffer("EventoTotalizador{");
+		sb.append("id=").append(id);
+		sb.append(", tipo='").append(tipo).append('\'');
+		sb.append(", nrReciboArquivoBase='").append(nrReciboArquivoBase).append('\'');
+		sb.append(", indApuracao=").append(indApuracao);
+		sb.append(", perApuracao='").append(perApuracao).append('\'');
+		sb.append(", cpfTrabalhador='").append(cpfTrabalhador).append('\'');
+		sb.append(", xmlEventoTotalizador='").append(xmlEventoTotalizador).append('\'');
+		sb.append(", evento=").append(evento);
+		sb.append('}');
+		return sb.toString();
+	}
 }

--- a/src/esocial-jt-service/src/main/java/br/jus/tst/esocialjt/negocio/AtualizacaoProcessamentoServico.java
+++ b/src/esocial-jt-service/src/main/java/br/jus/tst/esocialjt/negocio/AtualizacaoProcessamentoServico.java
@@ -1,7 +1,16 @@
 package br.jus.tst.esocialjt.negocio;
 
-import br.jus.tst.esocialjt.comunicacaogov.*;
-import br.jus.tst.esocialjt.dominio.*;
+import br.jus.tst.esocialjt.comunicacaogov.ComunicacaoEsocialGov;
+import br.jus.tst.esocialjt.comunicacaogov.RetornoEvento;
+import br.jus.tst.esocialjt.comunicacaogov.RetornoEventoTotalizador;
+import br.jus.tst.esocialjt.comunicacaogov.RetornoLote;
+import br.jus.tst.esocialjt.comunicacaogov.RetornoProcessamento;
+import br.jus.tst.esocialjt.dominio.CodigoResposta;
+import br.jus.tst.esocialjt.dominio.EnvioEvento;
+import br.jus.tst.esocialjt.dominio.ErroProcessamento;
+import br.jus.tst.esocialjt.dominio.Estado;
+import br.jus.tst.esocialjt.dominio.EventoTotalizador;
+import br.jus.tst.esocialjt.dominio.Lote;
 import br.jus.tst.esocialjt.negocio.exception.ComunicacaoEsocialGovException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -9,6 +18,7 @@ import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import javax.transaction.Transactional;
+import java.sql.SQLException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Set;
@@ -95,7 +105,18 @@ public class AtualizacaoProcessamentoServico {
 				eventoTotalizador.setPerApuracao(eventoTot.getPerApuracao());
 				eventoTotalizador.setCpfTrabalhador(eventoTot.getCpfTrabalhador());
 				eventoTotalizador.setXmlEventoTotalizador(eventoTot.getXmlEventoTotalizador());
-				eventoTotalizadorServico.salvar(eventoTotalizador);
+				try {
+					eventoTotalizadorServico.salvar(eventoTotalizador);
+				} catch (Exception e) {
+                    LOGGER.error("Erro ao salvar evento totalizador: {}", e.getMessage());
+					if (e.getCause() instanceof SQLException) {
+						SQLException sqlException = (SQLException) e.getCause();
+						LOGGER.error("SQL State: {}", sqlException.getSQLState());
+						LOGGER.error("Error Code: {}", sqlException.getErrorCode());
+					}
+					LOGGER.error(eventoTotalizador.toString());
+					LOGGER.debug(e.getMessage(), e);
+				}
 			}
 		}
 	}

